xmtp/src/association.rs

```diff
@@ -0,0 +1,109 @@
+use crate::types::Address;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use xmtp_crypto::signature::{RecoverableSignature, SignatureError, SigningKey};
+use xmtp_crypto::utils::{self, eth_address};
+
+#[derive(Debug, Error)]
+pub enum AssociationError {
+    #[error("bad signature")]
+    BadSignature(#[from] SignatureError),
+    #[error("Association text mismatch")]
+    TextMismatch,
+    #[error(
+        "Address mismatch in Association: Provided:{provided_addr:?} != signed:{signing_addr:?}"
+    )]
+    AddressMismatch {
+        provided_addr: Address,
+        signing_addr: Address,
+    },
+    #[error("unknown association error")]
+    Unknown,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+pub enum AssociationText {
+    Static { addr: Address, key_bytes: Vec<u8> },
```

Comment:
Always down for more clarity. Out of curiosity what is the source of the ambiguity?  I don't quote see it, which means there are likely other locations which would need the fix

Reply:
Updated to `(addr, account_public_key)` to be more descriptive. Does that help at all @richardhuaaa ?

I'd like to avoid `signing_address` if possible.